### PR TITLE
Add ability to support google manager tracker

### PR DESCRIPTION
**Pull request by jtapia**
### What it does?
- Add ability to specify what kind of tracker do you want to use on the admin
- Add script for google manager
- Remove unnecessary columns
- Add `tracker_type` to `spree_trackers` DB
- Refactor `current` method
- Add and fix test suite

### Preview
screencapture-localhost-4000-admin-stores-2-edit-2018-03-16-14_32_29

screencapture-localhost-4000-admin-trackers-2018-03-16-15_00_24

screencapture-localhost-4000-admin-trackers-new-2018-03-16-14_49_55

### CODE REVIEW

**Comment by fastjames** [comment]
https://app.clubhouse.io/enginecommerce/story/314/backend-to-manage-trackers

**Review by nspinazz89 — APPROVED** [review]
🍰 looks good

**Comment by aneekam** [comment]
https://app.clubhouse.io/enginecommerce/story/2085/add-ability-to-support-google-manager-tracker

**Comment by fastjames** [comment]
@jtapia can you squash this down?

File: Gemfile [hunk 1]
```diff
@@ -1,18 +1,39 @@
...
 gem 'solidus', github: 'solidusio/solidus', branch: branch
+gem 'solidus_auth_devise', '~> 1.0'
 
-gem 'rails-controller-testing', group: :test
-if branch == 'master' || branch >= "v2.3"
-  gem 'rails', '~> 5.1.0' # HACK: broken bundler dependency resolution
-elsif branch >= "v2.0"
-  gem 'rails', '~> 5.0.0' # HACK: broken bundler dependency resolution
+if branch == 'master' || branch >= 'v2.3'
+  gem 'rails', '~> 5.1.4'
```
Comment (fastjames) [inline Gemfile]: This probably needs to be `>=` now, so that it will test against 5.2.2.

File: lib/solidus_trackers/factories.rb [hunk 2]
```diff
@@ -1,8 +1,7 @@
 FactoryBot.define do
   factory :tracker, class: Spree::Tracker do
-    name 'Google Analytics'
     analytics_id 'A100'
     active true
-    script '<script></script>'
+    tracker_type 'google_analytics'
```
Comment (fastjames) [inline lib/solidus_trackers/factories.rb]: These attrs should be enclosed in { }s.

**Review by fastjames — APPROVED** [review]
Tests pass, if we get this squashed we can merge it.

File: Gemfile (hunk 1 above)
Comment (jtapia) [reply Gemfile]: done https://github.com/geminimvp/solidus_trackers/pull/6/commits/86c9ae54163e9f447526cd8fdca6c3e050d447da#diff-8b7db4d5cc4b8f6dc8feb7030baa2478R10

File: lib/solidus_trackers/factories.rb (hunk 2 above)
Comment (jtapia) [reply lib/solidus_trackers/factories.rb]: done https://github.com/geminimvp/solidus_trackers/pull/6/files#diff-c5de0a5e96f60b0e177c1d89ed4de2ffR5

**Comment by jtapia** [comment]
@fastjames the test build is passing, let me know if you need more changes here

**Comment by fastjames** [comment]
Looks good, can you squash it down?

**Comment by fastjames** [comment]
I have this squashed down now.

**Comment by fastjames** [comment]
Merged, :100:

**Comment by jtapia** [comment]
@fastjames sorry, I was attending other comments, thanks

**Pull request closed without merging**